core-rust/state-manager/src/staging/sorted_kv_merge_iterator.rs

```diff
@@ -0,0 +1,116 @@
+use std::cmp::Ordering;
+use std::iter::Peekable;
+
+/// Merges two ordered (by K) iterators of (K, V) pairs.
+/// Preserves the order by interleaving items from both iterators.
+/// If the same key is present in both iterators,
+/// a pair from the left iterator is returned
+/// and a pair from the right iterator is discarded.
+pub struct SortedKvMergeIterator<K, V, L, R>
```

Comment:
Really nice abstraction :D .

I might slightly prefer them called "overlay" and "base" or something rather than "left" and "right" because l/r doesn't suggest any semantics to me.
Also because I'd naturally put base in left and overlay in right :P